docs/enhancements/dynamic-fetching-of-openshift-versions.md

```diff
@@ -0,0 +1,280 @@
+            "cpu_architecture": "arm64",
+            "channels": ["candidate", "stable"]
+         },
+         {
+            "cpu_architecture": "multi",
+            "channels": ["candidate", "stable"]
+         }
+      ]
+   }
+}'
+```
+
+The assisted-service periodically polls versions from the ``upgrades_info`` API
+in the form of ``https://api.openshift.com/api/upgrades_info/v1/graph?channel=<channel>&arch=<arch>``.
+For example:
+```
+$ curl -s 'https://api.openshift.com/api/upgrades_info/v1/graph?channel=stable-4.11&arch=amd64' \
+    | jq -r '.nodes[] | select(.version | startswith("4.11")) | .version'
+4.11.9
+4.11.4
+4.11.8
+...
+```
+
+It then adds any new version found by this process in the database. When a
+consumer of the API requests the available versions via ``/v2/openshift-versions``
+(either it's the UI or a direct REST request) versions are read from the
+database.
+
+To help fetching only certain versions via auto-completion like U/X, we should
+provide a parameter ``version_pattern`` for the ``/v2/openshift-versions``
+endpoint. For example:
+```
+/v2/openshift-versions:
+   get:
+      ...
+      parameters:
+        - in: query
+          name: version_pattern
+          description: Version pattern for the returned releases. E.g. '4.12'.
+          type: string
+          required: false
+```
+
+### User Stories
+
+#### Installing The Latest Version of a z-stream
+
+As a user of assisted-installer UI, I should have a list of the latest versions
+for each minor release.
+
+#### Installing an Older Version of a z-stream
+
+As a user of assisted-installer UI, I should have the option to browse the
+different available versions that are fetched from upgrade channels.
+
+#### Configuring the Dynamically-loaded Versions of a Stream
+
+As a system administrator for the assisted-installer, I should have the option
+to configure both dynamically-loaded and statically-loaded OpenShift version.
+To do that, I can provide the following environment variables for example:
+```
+RELEASE_IMAGES='[
+  {
+    "openshift_version": "4.12-multi",
+    "cpu_architecture": "multi",
+    "cpu_architectures": [
+        "x86_64",
+        "arm64",
+        "ppc64le",
+        "s390x"
+    ],
+    "url": "quay.io/openshift-release-dev/ocp-release:4.12.0-rc.4-multi",
+    "version": "4.12.0-rc.4"
+  }
+]'
+
+DYNAMIC_OPENSHIFT_RELEASES_CONFIG='[
+  {
+    "openshift_version": "4.11",
+    "upgrade_channels": [
+        {
+          "cpu_architecture": "x86_64",
+          "channels": ["stable"]
+        }
+    ]
+  },
+  {
+    "openshift_version": "4.11",
+    "upgrade_channels": [
+      {
+        "cpu_architecture": "x86_64",
+        "channels": ["candidate", "stable"]
+      },
+      {
+        "cpu_architecture": "arm64",
+        "channels": ["candidate", "stable"]
+      }
+    ]
+  }
+]'
+```
+
+### Implementation Details/Notes/Constraints [optional]
+
+Mostly conveyed in ``Risks and Mitigations``.
+
+### Risks and Mitigations
```

Comment:
please add a case when we will get a new major release from the API, for example: 4.13 - we need to add it also to the image service repo

Reply:
please add a few words about the connection between assisted service versions and image service versions

Reply:
added